digitalocean/vpcpeering/resource_vpcpeering.go

```diff
@@ -0,0 +1,204 @@
+	vpcPeering, _, err := client.VPCs.CreateVPCPeering(context.Background(), vpcPeeringRequest)
+	if err != nil {
+		return diag.Errorf("Error creating VPC Peering: %s", err)
+	}
+
+	d.SetId(vpcPeering.ID)
+	log.Printf("[INFO] VPC Peering created, ID: %s", d.Id())
+
+	return resourceDigitalOceanVPCPeeringRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanVPCPeeringUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	if d.HasChange("name") {
+		vpcPeeringUpdateRequest := &godo.VPCPeeringUpdateRequest{
+			Name: d.Get("name").(string),
+		}
+
+		_, _, err := client.VPCs.UpdateVPCPeering(context.Background(), d.Id(), vpcPeeringUpdateRequest)
+		if err != nil {
+			return diag.Errorf("Error updating VPC Peering: %s", err)
+		}
+		log.Printf("[INFO] Updated VPC Peering")
+	}
+
+	return resourceDigitalOceanVPCPeeringRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanVPCPeeringDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	vpcPeeringID := d.Id()
+
+	err := retry.RetryContext(ctx, d.Timeout(schema.TimeoutDelete), func() *retry.RetryError {
+		resp, err := client.VPCs.DeleteVPCPeering(context.Background(), vpcPeeringID)
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusForbidden {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(fmt.Errorf("error deleting VPC Peering: %s", err))
+		}
+
+		if err := waitForVPCPeeringDeletion(ctx, client, vpcPeeringID, d.Timeout(schema.TimeoutDelete)); err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		d.SetId("")
+		log.Printf("[INFO] VPC Peering deleted, ID: %s", vpcPeeringID)
+
+		return nil
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceDigitalOceanVPCPeeringRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	vpcPeering, resp, err := client.VPCs.GetVPCPeering(context.Background(), d.Id())
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			log.Printf("[DEBUG] VPC Peering (%s) was not found - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading VPC Peering: %s", err)
+	}
+
+	d.SetId(vpcPeering.ID)
+	d.Set("name", vpcPeering.Name)
+	d.Set("status", vpcPeering.Status)
+	d.Set("vpc_ids", vpcPeering.VPCIDs)
+	d.Set("created_at", vpcPeering.CreatedAt.UTC().String())
+
+	return nil
+}
+
+func validateVPCIDs(val []string) error {
+	if len(val) != 2 {
+		return fmt.Errorf("'vpc_ids' must contain exactly 2 VPC IDs, got %d", len(val))
+	}
+
+	vpcIDMap := make(map[string]struct{})
+	for _, id := range val {
+		vpcID := id
+		if _, ok := vpcIDMap[vpcID]; ok {
+			return fmt.Errorf("'vpc_ids' must contain unique VPC IDs, duplicate found: %s", vpcID)
+		}
+		vpcIDMap[vpcID] = struct{}{}
+	}
+
+	return nil
+}
```

Comment:
not sure if we need this here since we already have serverside check..

Reply:
I added this to spot this issue before the request with wrong parameters is done, but probably not necessary. As you mentioned it is already handled serverside.

Reply:
I will remove this validation

Reply:
tbh... I am also not sure .. I'll let other's decide

Reply:
I have validated and imo this check isn't necessary. Since `TypeSet` can't have a `ValidateFunc` we will get this error from serverside. I mean, for other fields where we have the `ValidateFunc` it is for running `terraform validate` and return the errors before making an apply, but I think this can't be done for `TypeSet` or `TypeList`